assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -214,6 +214,47 @@ const AddressForm = ( {
 					);
 				}
 
+				const customValidationHandler = (
+					inputObject: HTMLInputElement
+				) => {
+					if (
+						! isPostcode( {
+							postcode: values.postcode,
+							country: values.country,
+						} )
+					) {
+						inputObject.setCustomValidity(
+							__(
+								'Please provide a valid postcode',
+								'woo-gutenberg-products-block'
+							)
+						);
+						return false;
+					}
+					return true;
+				};
+
+				if ( field.key === 'postcode' ) {
```

Comment:
See my other comments. I think we can just adjust the existing `ValidatedTextInput` below this block of code rather than redefine it for `postcode`. All that changes is the `customValidation`.